json-schema-registry/src/main/java/io/micronaut/jsonschema/registry/SchemaRegistryClientAuthFilter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jsonschema.registry;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.annotation.ClientFilter;
+import io.micronaut.http.annotation.RequestFilter;
+
+/**
+ * A client filter that adds basic auth to all requests.
+ *
+ * @since 1.4.0
+ * @author Elif Kurtay
+ */
+@ClientFilter("/**")
+@Requires(beans = SchemaRegistryConfig.class)
+@Requires(property = "registry.username")
+@Requires(property = "registry.password")
+public class SchemaRegistryClientAuthFilter {
```

Comment:
I don't think we should apply this globally. You can use a `@FilterMatcher` to narrow it to specific clients